store/mongo/datastore_mongo.go

```diff
@@ -740,6 +740,45 @@ func (db *DataStoreMongo) GetFiltersAttributes(ctx context.Context) ([]model.Fil
 	return attributes, nil
 }
 
+func (db *DataStoreMongo) DeleteGroup(
+	ctx context.Context,
+	group model.GroupName,
+) (chan model.DeviceID, error) {
+	deviceIDs := make(chan model.DeviceID)
+
+	database := db.client.Database(mstore.DbFromContext(ctx, DbName))
+	collDevs := database.Collection(DbDevicesColl)
+
+	filter := bson.M{DbDevAttributesGroupValue: group}
+	findOptions := &mopts.FindOptions{
+		Projection: bson.M{DbDevId: 1},
+	}
+	cursor, err := collDevs.Find(ctx, filter, findOptions)
+	if err != nil {
+		return nil, err
+	}
+	go func() {
+		defer cursor.Close(ctx)
+
+		update := bson.M{"$unset": bson.M{DbDevAttributesGroup: 1}}
+		for cursor.Next(ctx) {
+			device := &model.Device{}
+			if err = cursor.Decode(&device); err == nil {
+				updateDeviceID := model.DeviceID("")
+				res, err := collDevs.UpdateOne(ctx, bson.M{DbDevId: device.ID}, update)
+				if err == nil && res.MatchedCount == 1 {
+					updateDeviceID = device.ID
+				}
+				deviceIDs <- updateDeviceID
+			}
+		}
+
+		close(deviceIDs)
+	}()
```

Comment:
Yikes! This is not very efficient. We should also use batch operations (`UpdateMany`) for mongo as well, since we expect all the device IDs to match a single document (even though it's not necessarily always the case).

Reply:
My problem with `UpdateMany` is that we won't get back the device IDs, which means that we cannot reindex them in ES (reporting service).